### PR TITLE
[LLM]Arc starcoder example

### PR DESCRIPTION
**Pull request by cyita**
## Description

Add arc starcoder example.

### 4. How to test?
- [ ] Unit test
- [x] Manual test

### CODE REVIEW

File: python/llm/example/transformers/transformers_int4/GPU/starcoder/readme.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+```
+python ./generate.py --repo-id-or-model-path REPO_ID_OR_MODEL_PATH --prompt PROMPT --n-predict N_PREDICT
+```
+
+Arguments info:
+- `--repo-id-or-model-path REPO_ID_OR_MODEL_PATH`: argument defining the huggingface repo id for the StarCoder model to be downloaded, or the path to the huggingface checkpoint folder. It is default to be `'bigcode/starcoder'`.
+- `--prompt PROMPT`: argument defining the prompt to be infered (with integrated prompt format for chat). It is default to be `'def print_hello_world():'`.
+- `--n-predict N_PREDICT`: argument defining the max number of tokens to predict. It is default to be `32`.
+
+> **Note**: When loading the model in 4-bit, BigDL-LLM converts linear layers in the model into INT4 format. In theory, a *X*B model saved in 16-bit will requires approximately 2*X* GB of memory for loading.
```
Comment (rnwang04) [inline python/llm/example/transformers/transformers_int4/GPU/starcoder/readme.md:42]: It seems that we discard this part in other examples.

**Review by rnwang04 — APPROVED** [review]
others LGTM.

File: python/llm/example/transformers/transformers_int4/GPU/starcoder/readme.md (hunk 1 above)
Comment (cyita) [reply python/llm/example/transformers/transformers_int4/GPU/starcoder/readme.md]: Removed

**Comment by cyita** [comment]
### ENV
Arc-02
```
bigdl-core-xe             2.4.0b20230827           pypi_0    pypi
bigdl-llm                 2.4.0b20230827           pypi_0    pypi
```

### Performance
`starcoder-15.5b` @jason-dai 

| method    | 1st token, 32-32 | 2+ token, 32-32 | infer time | GPU peak memory usage(MB) | 1st token, 1024-128 | 2+ token, 1024-128 | infer time | GPU peak memory usage(MB) |
|-----------|------------------|-----------------|------------|---------------------------|---------------------|--------------------|------------|---------------------------|
| q4_0      | 373.3            | 49.9            | 1.9207s    | 10697                     | 1187                | 51                 | 7.6740s    | 13377                     |
| q4_0+half | 366.7            | 68.8            | 2.5098s    | 10041                     | 1158.4              | 70.1               | 10.0654s   | 11765                     |

**Pull request merged**